app/models/goal.py

```diff
@@ -2,5 +2,22 @@
 from app import db
 
 
+
+
+
+
+
 class Goal(db.Model):
+    __tablename__= "goals"
     goal_id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String(240))
+    tasks = db.relationship("Task", lazy=True, backref="goal")
+
+    def to_json(self):
+        if len(self.tasks) == 0: 
+            return {"id": self.goal_id, 
+            "title": self.title }
+        else:
+            return {"id": self.goal_id, 
+                    "title": self.title, 
+                    "tasks": self.tasks}
```

Comment:
Style: A couple of small things.

Since empty lists are falsey we can just check them as conditions.

Also, if you are going to split a dictionary over multiple lines the opening and closing braces should be on different lines from the key/value pairs.

```suggestion
        if not self.tasks: 
            return {
                "id": self.goal_id, 
                "title": self.title
            }
        else:
            return {
                "id": self.goal_id, 
                "title": self.title, 
                "tasks": self.tasks
            }
```